eip-4626.md

```diff
@@ -0,0 +1,152 @@
+
+## Motivation
+
+Yield bearing vaults have a lack of standardization leading to diverse implementation details. Some various examples include lending markets (Compound, Aave, Fuse), aggregators (Yearn, Rari Vaults, Idle), and intrinsically interest bearing tokens (xSushi). This makes integration difficult at the aggregator or plugin layer for protocols which need to conform to many standards. This forces each protocol to implement their own adapters which are error prone and waste development resources.
+
+A standard for yield bearing vaults will allow for a similar cambrian explosion to ERC-20, unlocking access to yield in a variety of applications with little specialized effort from developers.
+
+
+## Specification
+
+### Methods
+
+#### deposit
+
+`function deposit(address _to, uint256 _value) public returns (uint256 _shares)`
+
+Deposits `_value` tokens into the vault and grants ownership of them to `_to`.
+
+MAY return a pro-rata ownership `_shares` value corresponding `_value`, if not MUST return `0`.
+
+MAY represent `_shares` using internal accounting or an ERC-20 token. 
+
+If pro-rata shares ownership is implemented, the vault SHOULD implement `balanceOf`, `redeem`, `totalSupply` and `exchangeRate`.
+
+
+#### withdraw
+
+`function withdraw(address _to, uint256 _value) public returns (uint256 _shares)`
+
+Withdraws `_value` tokens from the vault and transfers them to `_to`. 
+
+MAY return a pro-rata ownership `_shares` value corresponding to `_value`,  if not MUST return `0`.
+
+
+#### totalHoldings
+`function totalHoldings() public view returns (uint256)`
+
+Returns the total amount of underlying tokens held/managed by the vault.
+
+
+#### balanceOfUnderlying
+`function balanceOfUnderlying(address _owner) public view returns (uint256)`
+
+Returns the total amount underlying tokens held in the vault for `_owner`.
+
+#### underlying
+`function underlying() public view returns (address)`
+
+Returns the address of the token the vault uses for accounting, depositing, and withdrawing. 
+
+SHOULD return a token implementing the ERC-20 standard.
+
+
+#### totalSupply
+
+`function totalSupply() public view returns (uint256)`
+
+Returns the total number of unredeemed vault shares in circulation.
+
+OPTIONAL - This method is only needed for vaults that implement a pro-rata share mechanism for deposits.
+
+#### balanceOf
+
+`function balanceOf(address _owner) public view returns (uint256)`
+
+Returns the total amount of vault shares the `_owner` currently has.
+
+OPTIONAL - This method is only needed for vaults that implement a pro-rata share mechanism for deposits.
+
+#### redeem
+
+`function redeem(address _to, uint256 _shares) public returns (uint256 _value)`
+
+Redeems a specific number of `_shares` for underlying tokens and transfers them to `_to`. 
+
+MAY return a pro-rata ownership `_shares` value corresponding `_value`, if not MUST return `0`.
```

Comment:
I think the documentation here is reversed, the return parameter here is rightfully value

Reply:
yup